integration_test/integration_test.go

```diff
@@ -298,6 +350,785 @@ func (t *WASMIntegrationTestSuite) Test04_ContractExecution() {
+
+		fmt.Println("Check the bonding status of the lower-powered general validator")
+		fmt.Println("Expected UNBONDED state")
+
+		val5Status, err := getValidatorBondingState(
+			desc.GetConnectionWithContext(context.Background()),
+			t.VolunteerValidatorWallet1.ByteAddress.Bytes(),
+		)
+		assert.NoError(t.T(), err)
+
+		if val5Status.String() == stakingtype.BondStatusUnbonded || val5Status.String() == stakingtype.BondStatusUnbonding {
+			fmt.Println("Lower-powered general validator is in unbonded status. Good")
+		} else {
+			fmt.Println("Lower-powered general validator is in bonded status. Test fail")
+			t.T().Fail()
+		}
+	}
+
+	{
+		fmt.Println("Try to turn off the volunteer validator")
+
+		cmd := exec.Command("docker", "stop", "xpla-localnet-volunteer3")
+		err := cmd.Run()
+		assert.NoError(t.T(), err)
+
+		fmt.Println("Volunteer validator down. Wait 5 blocktime to be jailed")
+		time.Sleep(time.Second*blocktime*jailBlocks + 1)
+		fmt.Println("Wait for validator list reorg..")
+		time.Sleep(time.Second*blocktime*validatorActiveBlocks + 1)
+
+		didVolunteerValVote, err := checkValidatorVoted(
+			desc.GetServiceDesc().ServiceConn,
+			t.VolunteerValidatorPVKey3.Address,
+		)
+		assert.NoError(t.T(), err)
+
+		if assert.False(t.T(), didVolunteerValVote) {
+			fmt.Println("Volunteer validator did not vote. Succeeded")
+		} else {
+			fmt.Println("Volunteer validator voted. Test fail")
+		}
+
+		didGeneralValVote, err := checkValidatorVoted(
+			desc.GetServiceDesc().ServiceConn,
+			t.VolunteerValidatorPVKey1.Address,
+		)
+		assert.NoError(t.T(), err)
+
+		if assert.True(t.T(), didGeneralValVote) {
+			fmt.Println("Lower-powered general validator voted. Succeeded")
+		} else {
+			fmt.Println("Lower-powered general validator did not voted. Test fail")
+		}
+
+		generalValStatus, err := getValidatorBondingState(
+			desc.GetConnectionWithContext(context.Background()),
+			t.VolunteerValidatorWallet1.ByteAddress.Bytes(),
+		)
+		assert.NoError(t.T(), err)
+
+		if assert.Equal(t.T(), stakingtype.BondStatusBonded, generalValStatus.String()) {
+			fmt.Println("Lower-powered general validator is in bonded status. Good")
+		} else {
+			fmt.Println("Lower-powered general validator is in unbonded status. Test fail")
+		}
+	}
+
+	{
+		fmt.Println("Turn on the volunteer validator and unjailing")
+
+		cmd := exec.Command("docker", "start", "xpla-localnet-volunteer3")
+		err := cmd.Run()
+		assert.NoError(t.T(), err)
+
+		fmt.Println("Wait enough time(30sec) to replay the blocks...")
+		time.Sleep(time.Second * 30)
+
+		unjailMsg := slashingtype.NewMsgUnjail(sdktypes.ValAddress(t.VolunteerValidatorWallet3.ByteAddress))
+
+		feeAmt := sdktypes.NewDec(xplaGeneralGasLimit).Mul(sdktypes.MustNewDecFromStr(xplaGasPrice))
+		fee := sdktypes.NewCoin(xplatypes.DefaultDenom, feeAmt.Ceil().RoundInt())
+
+		txhash, err := t.VolunteerValidatorWallet3.SendTx(ChainID, unjailMsg, fee, xplaGeneralGasLimit, false)
```

Comment:
I think this unjail tx is not work. 
Because `downtime_jail_duration` is 600s. It means after jailed, need to 600s.